### PR TITLE
Handle all scenario results in DAML Script IDE client

### LINKED ISSUE #7002: Verify error handling in DAML Script service
Someone needs to go through all of `SResult` and `SError` and verify which errors can arise and how we handle them.

### PR DESCRIPTION
**Pull request by aherrmann-da**
Closes #7002 

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala [hunk 1]
```diff
@@ -441,16 +441,38 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
                   result = Success(Right(results.toSeq))
               }
           }
+        case SResultFinalValue(v) =>
+          // The final result should always be unit.
+          result = Failure(new RuntimeException(s"Unexpected non-unit final result: $v"))
```
Comment (cocreature) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:446]: I guess if we want to go for FATAL, this should also be FATAL?

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala [hunk 2]
```diff
@@ -441,16 +441,38 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
...
           result = Failure(
             new RuntimeException("FATAL: Encountered scenario commit in DAML Script"))
         case SResultError(err) =>
           // Capture the error and exit.
           result = Failure(err)
-        case err =>
-          // TODO: Figure out when we hit this
-          // Capture the error (but not as SError) and exit.
-          result = Failure(new RuntimeException(s"FAILED: $err"))
+        case SResultNeedTime(callback @ _) =>
```
Comment (cocreature) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:453]: What happens if you have a `getTime` call in a choice that you call via DAML Script? I would expect that you will hit this in this case. The one handled by the DAML Script runner is `getTime` outside of a transaction.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala [hunk 3]
```diff
@@ -441,16 +441,38 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
...
+          result = Failure(
+            new RuntimeException(
+              "FATAL: GetTime should have been handled by the DAML Script runner"))
+        case SResultNeedPackage(pkg, callback @ _) =>
+          result = Failure(
+            new RuntimeException(
+              s"FATAL: Missing package $pkg should have been reported at Script compilation"))
+        case SResultScenarioInsertMustFail(committers @ _, optLocation @ _) =>
```
Comment (cocreature) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:461]: It you want to be consistent the errors from here on could all start with `Encountered scenario instruction`.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala (hunk 1 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala]: Agreed, done.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala (hunk 3 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala:459]: Good point, I've made the messages more consistent.

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala (hunk 2 above)
Comment (aherrmann-da) [reply daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala]: Indeed, I've extended the time test-case and changed this bit to handle this case.

**Pull request merged**